django_datawatch/querysets.py

```diff
@@ -9,8 +9,11 @@
 
 class ResultQuerySet(models.QuerySet):
     def for_user(self, user):
-        return self.filter(Q(assigned_to_group__isnull=True) | Q(assigned_to_group__in=user.groups.all()),
-                           Q(assigned_to_user__isnull=True) | Q(assigned_to_user=user))
+        user_groups = user.groups.all()
+        return self.filter(
```

Comment:
Shouldn't this be
```python
return self.filter(
    Q(assigned_groups__isnull=True)  | Q(assigned_groups__in=user_groups),  # not assigned to a group or to one of mine
    Q(assigned_users__isnull=True) | Q(assigned_users=user),  # not assigned to a user of i'm one of the assignees
)

Reply:
About this, I am not sure if this is a valid case but I modified what you suggested to count for the case if the user is part of the assigned users but not part of the assigned groups (assigned groups is not null)

Reply:
Might be that a user was in a group, got a result assigned and then gets removed from that group.  
They should then still be able to see that result.